qdrant-landing/content/documentation/guides/running-with-GPU.md

```diff
@@ -0,0 +1,204 @@
+- Latest NVIDIA drivers
+- [nvidia-container-toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html)
+
+Most AI or CUDA images on Amazon/GCP come pre-configured with the NVIDIA container toolkit.
+
+### Docker images with NVidia GPU support
+
+Nvidia docker images have the same tag as Qdrant but with `gpu-nvidia` suffix, for example `qdrant/qdrant:v1.13.0-gpu-nvidia`.
+This docker image contains all required dependencies to run Qdrant with GPU support.
+
+To enable GPU for docker, you need an additional `--gpus=all` flag. To enable GPU for Qdrant you need to set enable flag from settings. Example of starting Qdrant with NVidia GPU support:
+
+```bash
+# `--gpus=all` flag says to Docker that we want to use GPUs.
+# `-e QDRANT__GPU__indexing=1` flag says to QDrant that we want to use GPUs for indexing.
+docker run \
+	--rm \
+	--gpus=all \
+	-p 6333:6333 \
+	-p 6334:6334 \
+	-e QDRANT__GPU__indexing=1 \
+	qdrant/qdrant:gpu-nvidia-latest
+```
+
+To ensure that the GPU was initialized correctly, you may check it in logs. First Qdrant prints all found GPU devices without filtering and then prints list of all created devices:
+
+```
+2025-01-13T11:58:29.124087Z  INFO gpu::instance: Foung GPU device: NVIDIA GeForce RTX 3090    
+2025-01-13T11:58:29.124118Z  INFO gpu::instance: Foung GPU device: llvmpipe (LLVM 15.0.7, 256 bits)    
+2025-01-13T11:58:29.124138Z  INFO gpu::device: Create GPU device NVIDIA GeForce RTX 3090    
+```
+
+In this example logs you may see that 2 devices were found: RTX 3090 and llvmpipe (a CPU-emulated GPU which is included into docker image). And later you may see that only RTX was initialized.
+
+That’s it. In a basic scenario, there is nothing to do more.
+
+### Troubleshoots
+
+If your GPU is not detected in docker, check first your driver and `nvidia-container-toolkit` versions.
+They both should be latest.
+
+To check that Vulkan API is visible in a docker container, try to run this command:
+
+```
+docker run --rm --gpus=all qdrant/qdrant:gpu-nvidia-latest vulkaninfo --summary
+```
+
+You may find an error message with an explanation of why the NVidia device is not visible.
+If your NVidia GPU is not visible in Docker, Docker image cannot use libGLX_nvidia.so.0 on your host. Example of an error message:
+
+```
+ERROR: [Loader Message] Code 0 : loader_scanned_icd_add: Could not get `vkCreateInstance` via `vk_icdGetInstanceProcAddr` for ICD libGLX_nvidia.so.0
+WARNING: [Loader Message] Code 0 : terminator_CreateInstance: Failed to CreateInstance in ICD 0. Skipping ICD.
+```
+
+To fix it, try to change the config:
+
+```
+sudo nano /etc/nvidia-container-runtime/config.toml
+```
+
+Set `no-cgroups=false`, save config, and restart docker
+
+```
+sudo systemctl restart docker
+```
+
+## AMD GPUs
+
+AMD docker images have the same tag as qdant but with `gpu-amd` suffix, for example `qdrant/qdrant:v1.13.0-gpu-amd`.
```

Comment:
AMD host should have ROCm. Can you please all that user may install ROCm. Link https://rocm.docs.amd.com/projects/install-on-linux/en/latest/install/detailed-install.html. And you also may mention like in NVidia part that server images have already all installed soflware